### PR TITLE
Fix filterering in `distinct` for Tigris handler

### LINKED ISSUE #1704: Enable support for arrays in Tigris
This task is a continuation of #908.

After having arrays in `tjson`, we can integrate them and write tests.

A possible approach could be the following:
- Either creation of a collection or the first insertion of array should define the schema for a particular array field.
- If validation doesn't pass on Tigris side (an array contains various data types, attempt to insert an array of a type different than the one defined before etc), an error should be returned to the client.

### Testing

- Positive test cases can be covered with compat tests.
- Negative test cases can't be written unless we have validation for Mongo collections (where we can enable the same schema rules as the ones we have for Tigris).
- Diff tests are not possible because they don't work with Tigris.

Test case to check manually:
1. I insert an array to a new collection to a field called `tags`: `db.things.insertOne( { tags: ["a", "b", "c"] } );`
2. As the array contains data of the same type, insertion pass. The schema should describe string-based array.
3. I try to insert another array to the same field: `db.things.insertOne( { tags: [1, 2, 3] } );`
4. I should receive an error as the data type doesn't match (I try to insert `int`, not `string`).

### Definition of done
- [x] Add a valid data item to `sharedata.Composites`, something like `DocumentsDocuments`, but for arrays.
- [x] Add compat tests to work with arrays. Try to reuse existing array-related compat tests and shared data. Cases to cover (see TODOs in the code):
  - [x] create;
  - [x] insert, upsert, update;
  - [x] queries - array operations.
- [x] Add negative test cases (invalid or empty arrays) at least for unit tests.

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

Closes #1704.

ℹ️ This PR doesn't cover the main task but I sent it separately because it fixes a particular bug in distinct (`filter` wasn't applied to it). 

Spot refactoring: introduced `fetchAndFilterDocs` method for TIgris handler, similar to what we use for Postgres.

I will provide separate PRs to cover more cases.

## Readiness checklist

* [x] I added unit tests for new functionality or bug fixes.
* [ ] I added integration tests for new functionality or bug fixes.
* [x] I added compatibility tests for new functionality or bug fixes.
* [x] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by chilagrow — APPROVED** [review]
Thanks for the refactoring too :muscle:

File: internal/handlers/tigris/msg_update.go [hunk 1]
```diff
@@ -91,6 +91,9 @@ func (h *Handler) MsgUpdate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		if q, err = common.GetOptionalParam(update, "q", q); err != nil {
 			return nil, err
 		}
+
+		fp.Filter = q
```
[the remark this answers is not in the document]
Comment (rumyantseva) [reply internal/handlers/tigris/msg_update.go:95]: Yeah, and same for distinct. So for this one, we didn't use query pushdown even when possible, and for distinct, we didn't filter at all. Fixed now! :)

**Review by w84thesun — APPROVED** [review]
I like that schema properties thing!
